metron-platform/metron-solr/metron-solr-common/README.md

```diff
@@ -164,3 +164,142 @@ The `create_collection.sh` script depends on schemas installed in `$METRON_HOME/
+'2018-01-14T21:00:00:00', 'NOW/SECOND', 'NOW/DAY'
+
+    * ROUTER_FIELD: The name of the field in the incoming document that contains the datetime to route on - field must be of SOLR type DateTrie or DatePoint. 
+    For METRON this is standardised as field `datetime`.
+    
+    * ROUTER_INTERVAL: SOLR Date math format. The interval of time that each collection holds. eg "+1DAY", "+6HOUR", "+1WEEK" (`+` must be URL encoded to `%2B` )
+    
+    * ROUTER_MAXFUTUREMS: Optional field containing the number of milliseconds into the future that it is considered valid to have an event time for.
+    Documents with an event time exceeding this time period in the future are considered invalid and an error is returned.  Used as a sanity check to prevent 
+    the creation of unnecessary collections due to corrupted datetime values in events. Defaults is to ignore anything more then 10 minutes into the future.
+    
+    * ROUTER_AUTODELETEAGE: Optional field in SOLR Date math format. If this field is present, any time a collection is created, 
+    the oldest collections are assessed for deletion. Collections are deleted if the datetime interval they represent is older then
+     NOW - AUTODELETE_INTERVAL.  eg -2WEEK, -3MONTH, -1YEAR.  (`-` is a valid URL character that does not need to URL encoded.) 
+    
+    * CONFIGSET: Name of the collection configset that was created in the previous step - this is used a template for new collections.
+    
+    * CREATE-COLLECTION.*: These allow for Create collection options (e.g. numShards or numReplicas) to be specified directly in the 
+    create alias command.  
+    
+    Then the following command will create a time-routed alias:
+    ```
+        curl http://$SOLR_HOST:8983/solr/admin/collections?action=CREATEALIAS\
+       &name=$ALIAS_NAME\
+       &router.start=$ROUTER_START\
+       &router.field=$ROUTER_FIELD\
+       &router.name=time\
+       &router.interval=$ROUTER_INTERVAL\
+       &router.maxFutureMs=$ROUTER_MAXFUTUREMS\
+       &create-collection.collection.configName=$CONFIGSET\
+       &create-collection.numShards=2
+    ```
+   
+   
+1. Add a Metron Parser Stellar field transformation to the parser config that adds a correctly formatted datetime string to the event as it is being parsed:
+    1. Set environment variables for later reference
+        ```
+        source /etc/defaults/metron
+        export HDP_HOME="/usr/hdp/current"
+        export PARSER_NAME=<The name of the relevant parser>
+        ```
+       
+    1. Pull the most recent sensor parser config from zookeeper
+        ```
+        ${METRON_HOME}/bin/zk_load_configs.sh -i ${METRON_HOME}/config/zookeeper -m PULL -c PARSER -n $PARSER_NAME -z $ZOOKEEPER
+        ```
+       
+    1. Open the file to the relevant sensor parser at `$METRON_HOME/config/zookeeper/parsers/$PARSER_NAME.json`
+    
+    1. Add to the sensor parser config json field the following transformation:
+        ```
+          "fieldTransformations" : [{
```

Comment:
I'm guessing the markdown changed some of the double quotes.. this should be something like....

```
{
  rserClassName":"org.apache.metron.parsers.bro.BasicBroParser",
  "sensorTopic":"bro",
  "parserConfig": {
  "fieldTransformations" : [{
      "input" : [ ]"parserClassName":"org.apache.metron.parsers.bro.BasicBroParser",
  "sensorTopic":"bro",
  "parserConfig": {
  "fieldTransformations" : [{
      "input" : [ ]
    ,"transformation" : "STELLAR"
    ,"output" : [ "datetime"  ]
    ,"config" : {
      "datetime" : "DATE_FORMAT(\"yyyy-MM-dd'T'HH:mm:ss.SSSX\",timestamp)"
     }
    }]
}
}
```